### PR TITLE
POC : Support path alias in tina.config

### PR DESCRIPTION
**Pull request by Ben0189**
The alias path in the ts config file is now able to resolve by esbuild for tina.config file 

**Summary of work done in this pr** 
- update esbuild packages, the latest esbuild version support the path alias from ts config

### CODE REVIEW

**Comment by kldavis4** [comment]
@Ben0189 what is the [esbuild-plugin-alias-config](https://www.npmjs.com/package/esbuild-plugin-alias-path) providing?

**Comment by Ben0189** [comment]
> @Ben0189 what is the [esbuild-plugin-alias-config](https://www.npmjs.com/package/esbuild-plugin-alias-path) providing?

@kldavis4 Initial plan was to use this lib to support the alias from tsconfig
image
Figure: resources from esbuild [community](https://github.com/esbuild/community-plugins)

But looks like latest esbuild version does that for us

File: packages/@tinacms/cli/package.json [hunk 1]
```diff
@@ -78,7 +78,7 @@
 		"cors": "^2.8.5",
 		"crypto-js": "^4.2.0",
 		"dotenv": "^16.4.5",
-		"esbuild": "^0.18.20",
```
Comment (ncn-ssw) [inline packages/@tinacms/cli/package.json:81]: **question**: Should the version of `esbuild` included here match the 1 in the [[at]tinacms/scripts](https://github.com/tinacms/tinacms/blob/67b87d8bbcdc6ff660194869e72d2a204e0d5d93/packages/%40tinacms/scripts/package.json#L25) package?

Comment (Ben0189) [reply packages/@tinacms/cli/package.json:81]: Done update the scripts packages to use the same version of esbuild

File: examples/next-2024/tsconfig.json [hunk 2]
```diff
@@ -17,10 +21,21 @@
         "name": "next"
       }
     ],
+    "baseUrl": ".",
     "paths": {
-      "@/*": ["./*"]
+      "@lib/*": ["lib/*"],
+      "@components/header": ["components/header.ts"],
```
Comment (ncn-ssw) [inline examples/next-2024/tsconfig.json:27]: **question**: Are all these path aliases still in use?

File: examples/tina-self-hosted-demo/tina/tina-lock.json
Comment (ncn-ssw) [inline examples/tina-self-hosted-demo/tina/tina-lock.json:1]: **question**: Why delete the whole `tina-lock.json` file?

Comment (Ben0189) [reply examples/tina-self-hosted-demo/tina/tina-lock.json:1]: ah was to revert the changes to that file, but accidentally removed (using github UI)

Revert the commit, the file is now back

File: examples/next-2024/tsconfig.json (hunk 2 above)
Comment (Ben0189) [reply examples/next-2024/tsconfig.json]: Done remove all the path, the ts config is already use in another example so we no longer need this

**Review by ncn-ssw — APPROVED** [review]
🦙 Modified example runs correctly

**Pull request merged**